plugins/modules/site_workflow_manager.py

```diff
@@ -1198,6 +1199,155 @@ def get_want(self, config):
+            'latitude', 'longitude', and other site-specific details depending on the site type (area, building, or floor).
+        """
+        errormsg = []
+
+        if not config:
+            errormsg.append("Config data is missing.")
+        else:
+            for entry in config:
+                site = entry.get("site", {})
+                site_type = entry.get("type")
+                name = site.get(site_type, {}).get("name")
+                parent_name = site.get(site_type, {}).get("parent_name")
+                if name:
+                    param_spec = dict(type="str", length_max=40)
+                    validate_str(name, param_spec, "name", errormsg)
+                else:
+                    errormsg.append("parent_name should not be None or empty")
+
+                if parent_name:
+                    param_spec = dict(type="str", length_max=400)
+                    validate_str(parent_name, param_spec, "parent_name", errormsg)
+                else:
+                    errormsg.append("parent_name should not be None or empty")
+
+                if site_type:
+                    if site_type not in ("area", "building", "floor"):
+                        errormsg.append("site_type: Invalid value '{0}' for site_type in playbook. Must be one of: area, building, or Floor.".format(site_type))
+                else:
+                    errormsg.append("Site_type should not be None or empty")
+
+                if site_type == "building":
+                    address = site.get(site_type, {}).get("address")
+                    if address:
+                        param_spec = dict(type="str", length_max=255)
+                        validate_str(address, param_spec, "address", errormsg)
+
+                    latitude = site.get(site_type, {}).get("latitude")
+                    if latitude:
+                        if not (isinstance(latitude, (float, int)) and -90 <= latitude <= 90):
+                            errormsg.append("Invalid latitude, valid range is -90 to +90.")
+
+                    longitude = site.get(site_type, {}).get("longitude")
+                    if longitude:
+                        if not (isinstance(longitude, (float, int)) and -90 <= longitude <= 90):
+                            errormsg.append("Invalid longitude. Valid range is -90 to +90.")
+
+                    if not (latitude and longitude or address):
+                        errormsg.append("Either latitude/longitude or address is required.")
+                    elif (latitude and not longitude) or (not latitude and longitude):
+                        errormsg.append("Either Latitude or longitude is missing in the given playbook")
+
+                    country = site.get(site_type, {}).get("country")
+                    if country:
+                        param_spec = dict(type="str", length_max=100)
+                        validate_str(country, param_spec, "country", errormsg)
+                    else:
```

Comment:
self.log("Missing 'country' field in building entry.", "ERROR")

Reply:
Added